src/main/java/game/controller/FieldController.java

```diff
@@ -428,10 +430,90 @@ private boolean groupHasBuildings(int position) {
         return false;
     }
 
-    public boolean propertyCanBePawned(int position) {
-        if (!(fields[position] instanceof Property)) {
+    private boolean propertyCanBeSold(int position) {
+        if (fields[position].getField().equals("Street")) {
+            return groupHasBuildings(position);
+        }
+        return true;
+    }
+
+    public int[] sellablePropertyPositions(int player) {
+        int[] playerPropertyPositions = getPlayerPropertyPositions(player);
+        int[] sellablePropertyPositions = new int[0];
+        for (int position : playerPropertyPositions) {
+            if (propertyCanBeSold(position)) {
+                sellablePropertyPositions = Utility.addToArray(sellablePropertyPositions, position);
+            }
+        }
+        return sellablePropertyPositions;
+    }
+
+    private boolean buildingCanBeSold(int position) {
+        Property property = (Property) fields[position];
+        int propertyLevel = property.getPropertyLevel();
+        if (property instanceof Street && propertyLevel > 1) {
+            Property nextProperty = (Property) fields[property.getNextRelatedProperty()];
+            for (int i = 0; i < property.getRelatedProperties() - 1; i++) {
+                if (nextProperty.getPropertyLevel() > propertyLevel) {
+                    return false;
+                }
+                nextProperty = (Property) fields[nextProperty.getNextRelatedProperty()];
+            }
+
+        } else {
             return false;
         }
+        return true;
+    }
+
+    public int[] sellableBuildingPositions(int player) {
+        int[] playerPropertyPositions = getPlayerPropertyPositions(player);
+        int[] sellableBuildingPositions = new int[0];
+        for (int position : playerPropertyPositions) {
+            if (buildingCanBeSold(position)) {
+
+                sellableBuildingPositions = Utility.addToArray(sellableBuildingPositions, position);
+            }
+        }
+        return sellableBuildingPositions;
+    }
+
+    public int sellBuilding(int position) {
+        if (buildingCanBeSold(position)) {
+            Street street = (Street) fields[position];
+            street.setPropertyLevel(street.getPropertyLevel() - 1);
+            return street.getBuildingCost() / 2;
+        } else {
+            return 0;
+        }
+    }
+
+    public int sellAllPlayerProperties(int player) {
+        Property[] playerProperties = getPlayerProperties(player);
+        int totalValue = 0;
+        for (Property property : playerProperties) {
+            if (property.getField().equals("Street")) {
+
+                int propertyLevel = property.getPropertyLevel();
+                if (propertyLevel > 1) {
+                    Street street = (Street) property;
+                    totalValue += (street.getBuildingCost() / 2) * (propertyLevel - 1);
```

Comment:
I stedet for `propertyLevel - 1` så brug `street.getNumberOfBuildings()`, ellers vil den tro at der er 5 bygninger, når der er et hotel.

Reply:
sellAllPlayerProperties bliver kaldt hvis getPlayerTotalValue i Game returnerer en negativ værdi (altså at spilleren er helt fallit), derfor sælger den det hele på en gang. Når man sælger et hotel, så er der efterfølgende fire huse på grunden. Så det kan lige så godt betragtes som fem bygninger kodemæssigt, da man ellers først ville sælge hotellet og derefter de fire huse.